ci2/ci/github.py

```diff
@@ -136,9 +152,7 @@ def update_from_gh_json(self, gh_json):
         if self.source_sha != new_source_sha:
             log.info(f'{self.short_str()} source sha changed: {self.source_sha} => {new_source_sha}')
             self.source_sha = new_source_sha
-            self.sha = None
-            self.batch = None
-            self.build_state = None
+            self.set_build_state_to_unknown()
```

Comment:
I prefer something like `clear_build` since it just doesn't set the state, but the other associated build variables like sha and batch.